src/Workspaces/CSharp/Portable/Utilities/FormattingRangeHelper.cs

```diff
@@ -287,13 +287,18 @@ private static SyntaxToken GetPreviousTokenIfNotFirstTokenInTree(SyntaxToken tok
         }
 
         public static bool AreTwoTokensOnSameLine(SyntaxToken token1, SyntaxToken token2)
-        {
+        {           
             var tree = token1.SyntaxTree;
             if (tree != null && tree.TryGetText(out var text))
             {
                 return text.AreOnSameLine(token1, token2);
             }
 
+            if(token1.Equals(token2))
+            {
+                return token1.ToFullString().ContainsLineBreak();
```

Comment:
```suggestion
                return !token1.ToFullString().ContainsLineBreak();
```

Reply:
Wouldn't getting the [leading](http://source.roslyn.io/#Microsoft.CodeAnalysis/Syntax/SyntaxToken.cs,f0dc80731e1b9aec,references) and [trailing](http://source.roslyn.io/#Microsoft.CodeAnalysis/Syntax/SyntaxToken.cs,410) trivia and checking for if the kind is [EndOfLine](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Syntax/SyntaxKind.cs,2c91b4623bea6586,references) be more efficient?

Reply:
i'm confused  if the tokesn are the same, then they're on the same line by definition.  So this cahnge seems innapropriate.  What is actually trying to be solved here?